netket/graph/lattice.py

```diff
@@ -226,13 +282,27 @@ def __init__(
         new_nodes = {old_node: new_node for new_node, old_node in enumerate(old_nodes)}
         graph = _nx.relabel_nodes(graph, new_nodes)
 
+        self._coords = dicts_to_array(self._atoms, "r_coord")
+        self._lattice_dims = _np.expand_dims(self.extent, 1) * self.basis_vectors
+
         # Order node names
         nodes = sorted(graph.nodes())
         edges = list(graph.edges())
         graph = _nx.MultiGraph()
         graph.add_nodes_from(nodes)
         graph.add_edges_from(edges)
 
+        self._inv_dims = _np.linalg.inv(self._lattice_dims)
+        frac_positions = _np.matmul(self._coords, self._inv_dims) % 1
+        frac_positions = frac_positions - frac_positions // 1
+        int_positions = (
+            _np.power(10, tol_digits) * _np.around(frac_positions, 5)
+        ).astype(int)
```

Comment:
this assumes `tol_digits == 5`
```suggestion
        int_positions = _np.around(frac_positions * 10**tol_digits)
```

Reply:
also I just think the power operator is perfectly good for our purposes